src/pages/home/report/ReportActionItem.js

```diff
@@ -19,30 +24,152 @@ const propTypes = {
+        this.capturePressLocation(nativeEvent);
+
+        // If the popover will display off-screen, scroll the ReportActionsView FlatList down to this item first.
+        if (this.popoverAnchorY - this.popoverHeight < 0) {
+            this.popoverAnchorY += (this.popoverHeight + 20);
+            this.props.scrollToThis(this.popoverHeight + 20);
+        }
+
+        if (!this.props.isSmallScreenWidth) {
+            // On large screens, only display the ReportActionContextMenu on RightClick, not LongPress.
+            if (isRightClick(nativeEvent)) {
+                this.setState({isModalVisible: true});
+            }
+        } else {
+            this.setState({isModalVisible: true});
+        }
+    }
+
+    /**
+     * Hide the ReportActionContextMenu modal popover.
+     */
+    hideModal() {
+        this.setState({isModalVisible: false});
+    }
+
+    /**
+     * Measure the size of the ReportActionContextMenu popover.
+     *
+     * @param {Object} nativeEvent
+     */
+    measurePopover({nativeEvent}) {
+        this.popoverWidth = nativeEvent.layout.width;
+        this.popoverHeight = nativeEvent.layout.height;
+    }
+
+    render() {
+        const {getContainerStyle, getModalStyleOverride} = getReportActionItemStyles();
+        return (
+            <PressableWithSecondaryInteraction onSecondaryInteraction={this.showModal}>
+                <Hoverable>
+                    {hovered => (
+                        <View>
+                            <View style={getContainerStyle(hovered)}>
+                                {!this.props.displayAsGroup
+                                    ? <ReportActionItemSingle action={this.props.action} />
+                                    : <ReportActionItemGrouped action={this.props.action} />}
+                            </View>
+                            <View style={styles.miniReportActionContextMenuWrapperStyle}>
+                                <ReportActionContextMenu
+                                    reportID={this.props.reportID}
+                                    reportActionID={this.props.action.sequenceNumber}
+                                    isVisible={hovered && !this.state.isModalVisible}
+                                    isMini
+                                />
+                            </View>
+                            <Modal
+                                type={CONST.MODAL.MODAL_TYPE.POPOVER}
+                                isVisible={this.state.isModalVisible}
+                                onClose={this.hideModal}
+                                styleOverride={getModalStyleOverride(
+                                    this.props.windowWidth,
+                                    this.props.windowHeight,
+                                    this.popoverAnchorX,
+                                    this.popoverAnchorY,
+                                    this.popoverWidth,
+                                    this.popoverHeight,
```

Comment:
Yeah, seeing these makes me think we should abstract this into its own `PopoverWithPosition` component. This might help untangle things a bit.

Reply:
Done!